### ISSUE TITLE
micromdm sometimes responses http 200 with commanduuid for v1/command request, but sometimes it responded http 201 with empty body

### ISSUE THREAD

**Issue by ppdouble** [micromdm/micromdm#905]
### What version of `micromdm` are you using?

micromdm v 1.9.0

### What `micromdm` command did you run?

ManagelApplicationList and InstallApplication

### What did you expect to see?

And the status could not change, when micromdm received v1/command request.

When I send v1/command, it should response the commandUUID, so that I could know who issued the command when I handle the corresponding mdm.Connect webhook.

### What did you see instead?

**When I use micromdm downloaded from github release page**

**normal: response 201 with empty response body**

```shell
Jul 19 10:33:30 fine-frog-3.localdomain micromdm.tmp[3314]: transport=http method=POST status=201 proto=HTTP/1.0 host=172.18.156.32 user_agent="Apache-HttpClient/4.5.14 (Java/1.8.0_361)" path=/v1/commands
Jul 19 10:33:30 fine-frog-3.localdomain micromdm.tmp[3314]: level=info msg="queued event for device" device_udid=A4A15689-CE1D-4823-95C8-40CF6ABE584B command_uuid=daed05c5-5765-4066-98c5-3d8601e9da46 request_type=InstallApplication
```

**with debug: response 200 with commanduuid**
```shell
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: ---BEGIN Request---
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: POST /v1/commands HTTP/1.0
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: Host: micromdm.publish.url
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: Connection: close
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: Accept: application/json;charset=utf-8
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: Accept-Encoding: gzip,deflate
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: Authorization: Basic cAljcm1tZG02dmRhb0xPSVpn8jFUQTZVYjRMa5VuKT03
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: Connection: close
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: Content-Length: 196
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: Content-Type: application/json;charset=utf-8
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: Remote_host: 172.18.156.32
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: User-Agent: Apache-HttpClient/4.5.14 (Java/1.8.0_361)
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: X-Forwarded-For: 172.18.156.32
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: X-Real-Ip: 172.18.156.32
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: {"management_flags":1,"request_type":"InstallApplication","manifest_url":"https://example.com/plusbeta.plist","udid":"A4A15689-CE1D-4823-95C8-40CF6ABE584B","change_management_state":"Managed"}
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: ---END Request---
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: ---BEGIN Response---
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: HTTP/1.1 201 Created
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: Connection: close
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: Content-Type: application/json; charset=utf-8
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: {
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: "payload": {
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: "command_uuid": "cfc6f7d3-e4dc-4a49-b151-dc44a5f7fe0c",
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: "command": {
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: "request_type": "InstallApplication",
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: "management_flags": 1,
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: "change_management_state": "Managed",
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: "manifest_url": "https://example.com/plusbeta.plist"
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: }
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: }
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: }
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: ---END Response---
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: transport=http method=POST status=200 proto=HTTP/1.0 host=172.18.156.32 user_agent="Apache-HttpClient/4.5.14 (Java/1.8.0_361)" path=/v1/commands
Jul 19 10:34:10 fine-frog-3.localdomain micromdm.tmp[3385]: level=info msg="queued event for device" device_udid=A4A15689-CE1D-4823-95C8-40CF6ABE584B command_uuid=cfc6f7d3-e4dc-4a49-b151-dc44a5f7fe0c request_type=InstallApplication
```

----

**When I built micromdm from micromdm v1.9.0 source code(adding more logs) by go 1.17**

**normal: response 201 with empty response body**

```shell
Jul 19 09:28:52 fine-frog-3.localdomain micromdm.tmp[32324]: msg="start setupWebhooks ============="
Jul 19 09:28:52 fine-frog-3.localdomain micromdm.tmp[32324]: msg=http://127.0.0.1:9999/webhook
Jul 19 09:28:52 fine-frog-3.localdomain micromdm.tmp[32324]: level=debug component=depsync msg="loaded DEP config" cursor=
Jul 19 09:28:52 fine-frog-3.localdomain micromdm.tmp[32324]: level=info component=depsync msg="waiting for DEP token to be added before starting sync"
Jul 19 09:28:52 fine-frog-3.localdomain micromdm.tmp[32324]: ts="2023/07/19 09:28:52" msg="micromdm/platform/apns/push.go: MakePushendpoint()====="
Jul 19 09:28:52 fine-frog-3.localdomain micromdm.tmp[32324]: ts="2023/07/19 09:28:52" msg="micromdm endpoint: \"GET\": \"/push/{udid}\")"
Jul 19 09:28:52 fine-frog-3.localdomain micromdm.tmp[32324]: ts="2023/07/19 09:28:52" msg="micromdm endpoint: \"POST\": \"/v1/push/{udid}\")"
Jul 19 09:28:52 fine-frog-3.localdomain micromdm.tmp[32324]: ts="2023/07/19 09:28:52" msg="micromdm/platform/command/new_command.go====="
Jul 19 09:28:52 fine-frog-3.localdomain micromdm.tmp[32324]: ts="2023/07/19 09:28:52" msg="POST v1/commands"
Jul 19 09:28:52 fine-frog-3.localdomain micromdm.tmp[32324]: level=info msg="serving insecure HTTP" addr=127.0.0.1:8080
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: ts="2023/07/19 09:29:38" msg="new_command: call decodeNewCommandRequest:commandUUID ====="
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: ts="2023/07/19 09:29:38" msg="micromdm/platform/command/new_command.go call NewCommand()====="
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: ts="2023/07/19 09:29:38" caller="micromdm/platform/command/new_command.go call NewCommand(), payload.CommandUUID:11104" msg="b36-2964-4301-a919-b7c5cc450101======"
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: httputil.go call EncodeJSONResponse()====
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: ts="2023/07/19 09:29:38" msg="micromdm/platform/command/new_command.go call StatusCode(), StatusCode:%!s(int=201)======"
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: httputil.go  EncodeJSONResponse(): StatusCode: %!s(int=201) ====
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: httputil.go  EncodeJSONResponse(): response: {%!s(*mdm.CommandPayload=&{11104b36-2964-4301-a919-b7c5cc450101 0xc000318140}) <nil>} ====
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: httputil.go  EncodeJSONResponse(): enc.Encode(): %!s(<nil>) ====
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: transport=http method=POST status=201 proto=HTTP/1.0 host=172.18.156.32 user_agent="Apache-HttpClient/4.5.14 (Java/1.8.0_361)" path=/v1/commands
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: level=info msg="queued event for device" device_udid=A4A15689-CE1D-4823-95C8-40CF6ABE584B command_uuid=11104b36-2964-4301-a919-b7c5cc450101 request_type=InstallApplication
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: micromdm/platform/apns/push.go: Push()====
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: from db: pushinfo.Token 76326373524762775247646d6869544c30554a716c6c52495579656c6b584e32====
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: from db: pushinfo.Push Magic F6EF0F3E-6D36-4F8A-BABE-B5BDF0345D3E====
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: from db: pushinfo.UDID A4A15689-CE1D-4823-95C8-40CF6ABE584B====
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: from db: pushinfo.MDMTopic com.apple.mgmt.External.e82962f9-3a7a-4b41-882d-d8ff2baef2f2====
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: call Push info.Token:76326373524762775247646d6869544c30554a716c6c52495579656c6b584e32====
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: call Push hex.DecodeString(info.Token):A4A15689-CE1D-4823-95C8-40CF6ABE584B====
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: ts="2023/07/19 09:29:38" msg="buford/push/service.go: call Push()===="
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: ts="2023/07/19 09:29:38" caller="buford/push/service.go:Push: push url: http://127.0.0.1:8989" msg="/3/device/76326373524762775247646d6869544c30554a716c6c52495579656c6b584e32===="
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: ts="2023/07/19 09:29:38" msg="buford/push/service.go: Push: apns req headers.Topic: ===="
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: component="postWebhookEvent==========" msg="postWebhookEvent=========="
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: transport=http method=PUT status=200 proto=HTTP/1.0 host=172.18.156.32 user_agent=Go-http-client/2.0 path="/mdm/connect"
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: component="postWebhookEvent==========" msg="postWebhookEvent=========="
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: transport=http method=PUT status=200 proto=HTTP/1.0 host=172.18.156.32 user_agent=Go-http-client/2.0 path="/mdm/connect"
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: ts="2023/07/19 09:29:38" msg="buford/push/service.go: Push: apns response StatusCode: %!s(int=200)===="
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: ts="2023/07/19 09:29:38" caller="buford/push/service.go: Final Request to Apple: resp.StatusCode:%!s(int=200), apns-id:4" msg="BD601E3-3EA0-44E6-9448-87C5700079E9===="
Jul 19 09:29:38 fine-frog-3.localdomain micromdm.tmp[32324]: ts="2023/07/19 09:29:38" msg="micromdm/platform/apns/push.go: Push(): result: 4BD601E3-3EA0-44E6-9448-87C5700079E9===="
```

**with debug: response 201 with empty body**

```shell
Jul 19 09:38:00 fine-frog-3.localdomain micromdm.tmp[334]: msg="start setupWebhooks ============="
Jul 19 09:38:00 fine-frog-3.localdomain micromdm.tmp[334]: msg=http://127.0.0.1:9999/webhook
Jul 19 09:38:00 fine-frog-3.localdomain micromdm.tmp[334]: level=debug component=depsync msg="loaded DEP config" cursor=
Jul 19 09:38:00 fine-frog-3.localdomain micromdm.tmp[334]: level=info component=depsync msg="waiting for DEP token to be added before starting sync"
Jul 19 09:38:00 fine-frog-3.localdomain micromdm.tmp[334]: ts="2023/07/19 09:38:00" msg="micromdm/platform/apns/push.go: MakePushendpoint()====="
Jul 19 09:38:00 fine-frog-3.localdomain micromdm.tmp[334]: ts="2023/07/19 09:38:00" msg="micromdm endpoint: \"GET\": \"/push/{udid}\")"
Jul 19 09:38:00 fine-frog-3.localdomain micromdm.tmp[334]: ts="2023/07/19 09:38:00" msg="micromdm endpoint: \"POST\": \"/v1/push/{udid}\")"
Jul 19 09:38:00 fine-frog-3.localdomain micromdm.tmp[334]: ts="2023/07/19 09:38:00" msg="micromdm/platform/command/new_command.go====="
Jul 19 09:38:00 fine-frog-3.localdomain micromdm.tmp[334]: ts="2023/07/19 09:38:00" msg="POST v1/commands"
Jul 19 09:38:00 fine-frog-3.localdomain micromdm.tmp[334]: level=info msg="serving insecure HTTP" addr=127.0.0.1:8080
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ---BEGIN Request---
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: POST /v1/commands HTTP/1.0
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Host: lihu.tech
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Connection: close
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Accept: application/json;charset=utf-8
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Accept-Encoding: gzip,deflate
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Authorization: Basic cAljcm1tZG02dmRhb0xPSVpn8jFUQTZVYjRMa5VuKT03
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Connection: close
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Content-Length: 192
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Content-Type: application/json;charset=utf-8
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Remote_host: 172.18.156.32
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: User-Agent: Apache-HttpClient/4.5.14 (Java/1.8.0_361)
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: X-Forwarded-For: 172.18.156.32
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: X-Real-Ip: 172.18.156.32
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: {"management_flags":1,"request_type":"InstallApplication","manifest_url":"https://example.com/plusbeta.plist","udid":"A4A15689-CE1D-4823-95C8-40CF6ABE584B","change_management_state":"Managed"}
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ---END Request---
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ts="2023/07/19 09:39:14" msg="new_command: call decodeNewCommandRequest:commandUUID ====="
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ts="2023/07/19 09:39:14" msg="micromdm/platform/command/new_command.go call NewCommand()====="
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ts="2023/07/19 09:39:14" caller="micromdm/platform/command/new_command.go call NewCommand(), payload.CommandUUID:38" msg="e78ae4-4da9-43fb-8f3f-06346fee1c1d======"
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: httputil.go call EncodeJSONResponse()====
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ts="2023/07/19 09:39:14" msg="micromdm/platform/command/new_command.go call StatusCode(), StatusCode:%!s(int=201)======"
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: httputil.go  EncodeJSONResponse(): StatusCode: %!s(int=201) ====
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: httputil.go  EncodeJSONResponse(): response: {%!s(*mdm.CommandPayload=&{38e78ae4-4da9-43fb-8f3f-06346fee1c1d 0xc0000e2140}) <nil>} ====
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: httputil.go  EncodeJSONResponse(): enc.Encode(): %!s(<nil>) ====
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ---BEGIN Response---
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: HTTP/1.1 201 Created
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Connection: close
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Content-Type: application/json; charset=utf-8
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: {
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: "payload": {
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: "command_uuid": "38e78ae4-4da9-43fb-8f3f-06346fee1c1d",
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: "command": {
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: "request_type": "InstallApplication",
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: "management_flags": 1,
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: "change_management_state": "Managed",
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: "manifest_url": "https://example.com/plusbeta.plist"
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: }
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: }
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: }
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: {
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: "payload": {
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: "command_uuid": "38e78ae4-4da9-43fb-8f3f-06346fee1c1d",
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: "command": {
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: "request_type": "InstallApplication",
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: "management_flags": 1,
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: "change_management_state": "Managed",
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: "manifest_url": "https://example.com/plusbeta.plist"
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: }
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: }
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: }
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ---END Response---
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: transport=http method=POST status=200 proto=HTTP/1.0 host=172.18.156.32 user_agent="Apache-HttpClient/4.5.14 (Java/1.8.0_361)" path=/v1/commands
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: level=info msg="queued event for device" device_udid=A4A15689-CE1D-4823-95C8-40CF6ABE584B command_uuid=38e78ae4-4da9-43fb-8f3f-06346fee1c1d request_type=InstallApplication
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: micromdm/platform/apns/push.go: Push()====
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: from db: pushinfo.Token 76326373524762775247646d6869544c30554a716c6c52495579656c6b584e32====
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: from db: pushinfo.Push Magic F6EF0F3E-6D36-4F8A-BABE-B5BDF0345D3E====
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: from db: pushinfo.UDID A4A15689-CE1D-4823-95C8-40CF6ABE584B====
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: from db: pushinfo.MDMTopic com.apple.mgmt.External.5fe4edcf-c12b-4096-8a8b-30fbd4b6aab4====
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: call Push info.Token:76326373524762775247646d6869544c30554a716c6c52495579656c6b584e32====
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: call Push hex.DecodeString(info.Token):A4A15689-CE1D-4823-95C8-40CF6ABE584B====
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ts="2023/07/19 09:39:14" msg="buford/push/service.go: call Push()===="
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ts="2023/07/19 09:39:14" caller="buford/push/service.go:Push: push url: http://127.0.0.1:8989" msg="/3/device/76326373524762775247646d6869544c30554a716c6c52495579656c6b584e32===="
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ts="2023/07/19 09:39:14" msg="buford/push/service.go: Push: apns req headers.Topic: ===="
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ---BEGIN Request---
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: PUT /mdm/connect HTTP/1.0
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Host: lihu.tech
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Connection: close
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Accept-Encoding: gzip
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Connection: close
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Content-Length: 291
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Mdm-Signature: ZGQzMTNmcXdlZnF3ZnYoS01NKiZZSEclJXl5NjU1dG0sLi5cLyc7W11dK18rfX17JydcfHx7O2xs
a2tscGRwbGRmdihLTU0qJllIRyUleXk2NTV0bSwuWkdRek1UTm1jWGRsWm5GM1puWW9TMDFOS2la
WlNFY2xKWGw1TmpVMWRHMHNMaTVjTHljN1cxMWRLMThyZlgxN0p5ZGNmSHg3TzJ4cwphMnRzY0dS
d2JHUm1kaWhMVFUwcUpsbElSeVVsZVhrMk5UVjBiU3d1Q2c9PQo=
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Remote_host: 172.18.156.32
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: User-Agent: Go-http-client/2.0
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: X-Forwarded-For: 172.18.156.32
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: X-Real-Ip: 172.18.156.32
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: <?xml version="1.0" encoding="UTF-8"?>
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: <plist version="1.0"><dict><key>Status</key><string>Idle</string><key>UDID</key><string>A4A15689-CE1D-4823-95C8-40CF6ABE584B</string></dict></plist>
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ---END Request---
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: component="postWebhookEvent==========" msg="postWebhookEvent=========="
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ---BEGIN Response---
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: HTTP/1.1 200 OK
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Connection: close
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: <?xml version="1.0" encoding="UTF-8"?>
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: <plist version="1.0"><dict><key>Command</key><dict><key>ChangeManagementState</key><string>Managed</string><key>ManagementFlags</key><integer>1</integer><key>ManifestURL</key><string>https://example.com/plusbeta.plist</string><key>RequestType</key><string>InstallApplication</string></dict><key>CommandUUID</key><string>38e78ae4-4da9-43fb-8f3f-06346fee1c1d</string></dict></plist>
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ---END Response---
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: transport=http method=PUT status=200 proto=HTTP/1.0 host=172.18.156.32 user_agent=Go-http-client/2.0 path="/mdm/connect"
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ---BEGIN Request---
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: PUT /mdm/connect HTTP/1.0
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Host: lihu.tech
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Connection: close
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Accept-Encoding: gzip
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Connection: close
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Content-Length: 538
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Mdm-Signature: ZGQzMTNmcXdlZnF3ZnYoS01NKiZZSEclJXl5NjU1dG0sLi5cLyc7W11dK18rfX17JydcfHx7O2xs
a2tscGRwbGRmdihLTU0qJllIRyUleXk2NTV0bSwuWkdRek1UTm1jWGRsWm5GM1puWW9TMDFOS2la
WlNFY2xKWGw1TmpVMWRHMHNMaTVjTHljN1cxMWRLMThyZlgxN0p5ZGNmSHg3TzJ4cwphMnRzY0dS
d2JHUm1kaWhMVFUwcUpsbElSeVVsZVhrMk5UVjBiU3d1Q2c9PQo=
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Remote_host: 172.18.156.32
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: User-Agent: Go-http-client/2.0
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: X-Forwarded-For: 172.18.156.32
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: X-Real-Ip: 172.18.156.32
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: <?xml version="1.0" encoding="UTF-8"?>
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: <plist version="1.0"><dict><key>CommandUUID</key><string>38e78ae4-4da9-43fb-8f3f-06346fee1c1d</string><key>Identifier</key><string>net.mock.mobile.bundleid123</string><key>RequestType</key><string>InstallApplication</string><key>State</key><string>Prompting</string><key>Status</key><string>Acknowledged</string><key>UDID</key><string>A4A15689-CE1D-4823-95C8-40CF6ABE584B</string></dict></plist>
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ---END Request---
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: component="postWebhookEvent==========" msg="postWebhookEvent=========="
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ---BEGIN Response---
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: HTTP/1.1 200 OK
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: Connection: close
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ---END Response---
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: transport=http method=PUT status=200 proto=HTTP/1.0 host=172.18.156.32 user_agent=Go-http-client/2.0 path="/mdm/connect"
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ts="2023/07/19 09:39:14" msg="buford/push/service.go: Push: apns response StatusCode: %!s(int=200)===="
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ts="2023/07/19 09:39:14" msg="buford/push/service.go: Final Request to Apple: resp.StatusCode:%!s(int=200), apns-id:CA6C0FBA-31B8-4C6B-9C8B-A702CB35DB71===="
Jul 19 09:39:14 fine-frog-3.localdomain micromdm.tmp[334]: ts="2023/07/19 09:39:14" msg="micromdm/platform/apns/push.go: Push(): result: CA6C0FBA-31B8-4C6B-9C8B-A702CB35DB71===="
```

----

Related issue [#406](https://github.com/micromdm/micromdm/issues/406), mentioned at [#issuecomment-829853005](https://github.com/micromdm/micromdm/issues/663#issuecomment-829853005), [#issuecomment-1502156864](https://github.com/micromdm/micromdm/pull/871#issuecomment-1502156864)

**Comment by korylprince** [comment]
Issue #406 was fixed by #906. Can you build from source (main branch) and see if this fixes your issue?

**Comment by ppdouble** [comment]
Yes, I built the main branch. The status code is 201 in both normal and debug mode. @korylprince 

But, for the seconde part:

> When I send v1/command, it should response the commandUUID, so that I could know who issued the command when I handle the corresponding mdm.Connect webhook.

In the past, in debug mode it responsed command uuid, which looks like:

```
{
    "payload": {
        "command_uuid": "38e78ae4-4da9-43fb-8f3f-06346fee1c1d",
        "command": {
            "request_type": "InstallApplication",
            "management_flags": 1,
            "change_management_state": "Managed",
            "manifest_url": "https://example.com/plusbeta.plist"
        }
    }
}
```

But it is gone in debug mode now. I actually need the command uuid in release mode to check who issued the v1/command when I receive webhook. So that I can record the action to trace errors, even the webhook failed to come.

I don't find the code that wraps the v1/command response. I've never use golang before I use micromdm. Could you give me some clues about this? 

https://github.com/micromdm/micromdm/blob/v1.9.0/pkg/httputil/httputil.go#L51

https://github.com/micromdm/go4/blob/7c4a9b165b09948c8d5485933cad651183a4affd/httputil/middleware.go#L70


Should I post a new issue for this part?

**Comment by korylprince** [comment]
@ppdouble I can't reproduce this. With micromdm built from main (current commit c7e70b9), whether `-http-debug` is given or not, I receive HTTP 201 with the response body (including the command uuid) as expected.

My guess is there's something in your setup (e.g. reverse proxy for MicroMDM, your code calling MicroMDM's API) that's dropping the response body when it's getting HTTP 201.

Can you try reproducing this with a curl command that connects directly to MicroMDM (i.e. no proxy in between)?

**Comment by ppdouble** [comment]
Yes, I tried both.  The problem is that I did not catch 201 status. It works now. Both in normal mode and debug mode gave Response 201 with commandUUID. It solved my question. You can close this issue now.
